tiledb/sm/filesystem/azure.h

```diff
@@ -310,14 +311,12 @@ class Azure {
   Status write(const URI& uri, const void* buffer, uint64_t length);
 
   /**
-   * Returns a reference to the Azure blob service client.
+   * Initializes the Azure blob service client and returns a reference to it.
    *
-   * Used for testing. Calling code should include the Azure SDK headers to make
+   * Calling code should include the Azure SDK headers to make
    * use of the BlobServiceClient.
    */
-  const ::Azure::Storage::Blobs::BlobServiceClient& client() const {
-    return *client_;
-  }
+  const ::Azure::Storage::Blobs::BlobServiceClient& client() const;
```

Comment:
The problem of this method is that it doesn't prevent to use client_. We should make a singleton class that would hide client_ from this class.

Reply:
Done.